### ISSUE TITLE
Splinter Cell: Yellow bar on right screen edge

### ISSUE THREAD

**Issue by thrimbor** [xqemu/xqemu#124]
Possibly related to #123.

"Splinter Cell" shows a yellow bar on the right edge of the screen. The area where the bar appears and the color are always the same. Also, the yellow bar in SC seems to allow you to look through the wall.

![sc5](https://user-images.githubusercontent.com/1339483/47593968-b96d0500-d979-11e8-8f56-510763644a18.png)

Tested on Arch Linux (GTX 970, nvidia 410.66-1) and Windows 7 (GeForce 310M).

More screenshots available [here](https://nextcloud.improved-madness.de/index.php/s/p3gsLQzXY2HNXt7)

**Comment by MayeulC** [comment]
Looks a bit similar to https://github.com/xqemu/xqemu/pull/76#issuecomment-404529101

**Comment by JayFoxRox** [comment]
Judging by the screenshot alone, this is likely the same problem as #123, which has already been analyzed.

I suggest to wait for a fix for #123 before doing any further work on this issue.

**Comment by JayFoxRox** [comment]
*This post might be moved to a separate issue soon*

I've looked at the *green* bar on the right side of the screen, once the current version of #131 is used.

The game clears the a 512x512 buffer in RGRBA:(15,240,0,0) using `glClear` in XQEMU.

It then draws the current bright regions of the scene framebuffer onto it.
This step is broken and leaves a 1 pixel edge on the right side, in RGRBA:(15,240,0,0).
The scene framebuffer is fine, so the draw errornously doesn't contain all the scene.
Left, top and bottom are fine at 0,0,0,255.
The (fixed by #131) GLSL `clipRegion[0]` for this is (0,0,512,512).

Relevant vertex shader for blit:

```glsl
#define compositeMat mat4(c[0x00], c[0x00+1], c[0x00+2], c[0x00+3])
#define modelViewMat0 mat4(c[0x08], c[0x08+1], c[0x08+2], c[0x08+3])
#define invModelViewMat0 mat4(c[0x0c], c[0x0c+1], c[0x0c+2], c[0x0c+3])

void main() {
/* Skinning mode 0 */
vec4 tPosition = (position * modelViewMat0).xyzw;
vec3 tNormal = (vec4(normal, 0.0) * invModelViewMat0).xyz;
tNormal = normalize(tNormal);
/* Texgen for stage 0 */
oT0.x = texture0.x;
oT0.y = texture0.y;
oT0.z = texture0.z;
oT0.w = texture0.w;
/* Texgen for stage 1 */
oT1.x = texture1.x;
oT1.y = texture1.y;
oT1.z = texture1.z;
oT1.w = texture1.w;
/* Texgen for stage 2 */
oT2.x = texture2.x;
oT2.y = texture2.y;
oT2.z = texture2.z;
oT2.w = texture2.w;
/* Texgen for stage 3 */
oT3.x = texture3.x;
oT3.y = texture3.y;
oT3.z = texture3.z;
oT3.w = texture3.w;
  oD0 = diffuse;
  oD1 = specular;
  oB0 = backDiffuse;
  oB1 = backSpecular;
  tPosition = position;
   oPos = invViewport * (tPosition * compositeMat);
   oPos.z = oPos.z * 2.0 - oPos.w;
  vtx.inv_w = 1.0 / oPos.w;
  oFog.xyzw = vec4(1.0);

  vtx.D0 = clamp(oD0, 0.0, 1.0) * vtx.inv_w;
  vtx.D1 = clamp(oD1, 0.0, 1.0) * vtx.inv_w;
  vtx.B0 = clamp(oB0, 0.0, 1.0) * vtx.inv_w;
  vtx.B1 = clamp(oB1, 0.0, 1.0) * vtx.inv_w;
  vtx.Fog = oFog.x * vtx.inv_w;
  vtx.T0 = oT0 * vtx.inv_w;
  vtx.T1 = oT1 * vtx.inv_w;
  vtx.T2 = oT2 * vtx.inv_w;
  vtx.T3 = oT3 * vtx.inv_w;
  gl_Position = oPos;
  gl_PointSize = oPts.x;
}
```

Relevant vertex constants / uniform:

![2018-11-07-173637_697x274_scrot](https://user-images.githubusercontent.com/360330/48145780-de377580-e2b3-11e8-9902-c269f0cf745a.png)

![2018-11-07-173438_761x19_scrot](https://user-images.githubusercontent.com/360330/48145793-e1cafc80-e2b3-11e8-836e-36111eb12d8d.png)

<details><summary>I'd love to provide more information, but the vertex values shown in apitrace are probably bogus (Click here to show anyway)</summary>

![2018-11-07-173343_876x423_scrot](https://user-images.githubusercontent.com/360330/48145866-10e16e00-e2b4-11e8-8ab8-7f3ed508da3e.png)
![2018-11-07-173407_469x424_scrot](https://user-images.githubusercontent.com/360330/48145865-1048d780-e2b4-11e8-9589-782654882339.png)
</details><br>

This 512x512 image is then downsized (also by drawing again) in steps, to 256x256, 128x128, 64x64, down to 32x32. That 32x32 image is then blended onto the scene. So the effect is similar to that in #123.
